app/assets/javascripts/discourse/app/components/discourse-topic.js

```diff
@@ -56,7 +56,7 @@ export default Component.extend(Scrolling, MobileScrollDirection, {
 
   _hideTopicInHeader() {
     this.appEvents.trigger("header:hide-topic");
```

Comment:
I found that we trigger that same event in https://github.com/discourse/discourse/blob/f84eda7c8d68075b5a7b8a23c4472cdd9a645b77/app/assets/javascripts/discourse/app/routes/topic.js#L392, and we set the topic to `null` just afterwards. Should we set `header.inTopic` to `false` too here? If that’s the case, I was wondering whether we should rely on events to set `inTopic`? (and sorry if those are dumb questions :sweat_smile:)

Reply:
Thats a good question... Since the header service never came into play there (previously), and we haven't noticed any unintended behavior, I am not sure we need to add it from a functional perspective. For consistency, we _could_, but I don't think it is necessary.